src/Discovery/DiscoveryDetails.tsx

```diff
@@ -282,6 +288,24 @@ const DiscoveryDetails = (props: Props) => {
               </Button>
             )
             : null}
+            {(
+              props.user.username
+              && !userHasMethodForServiceOnResource('access', 'study_registration')
```

Comment:
This is not the right way to call `userHasMethodForServiceOnResource`

```suggestion
              && !userHasMethodForServiceOnResource('access', 'study_registration',
            props.modalData[studyRegistrationConfig.studyRegistrationAccessCheckField],
            props.userAuthMapping)
```

Reply:
And remove L294

Reply:
Fixed per discussion